libs/pika/execution/include/pika/execution/algorithms/require_started.hpp

```diff
@@ -453,7 +453,22 @@ namespace pika {
             {
                 return detail::partial_algorithm<require_started_t>{};
             }
-        } require_started{};
+        };
+
+        /// \anchor require_started
+        /// \brief Diagnose if a sender has not been started and terminates on destruction. It
+        /// forwards the values of the predecessor sender.
+        ///
+        /// Sender adaptor that takes any sender and returns a sender.
+        ///
+        /// The destructor is by default terminating if the sender has not been connected or if the
+        /// operation state has not been started.
+        /// The operation state of a @ref require_started sender is allowed to not be started, but it
+        /// has to be explicitly asked for with discard.
```

Comment:
```suggestion
        /// The destructor of the sender terminates it has not been connected. The destructor of the corresponding operation state also terminates if it has not been started.
        /// The @ref require_started sender sender is allowed to not be started if it has been explicitly requested with the @p discard member function.
```

Side note: I'm really not sure how we best document the `discard` member function. The sender type is not public... but we'd still like to document it somehow. Do you think it's enough the way you've mentioned it above?

Reply:
Yeah, I think the example is good, and no ideas at the moment about how to document the member function of an unspecified class...